generator/js_generator.cc

```diff
@@ -3637,8 +3637,13 @@ void Generator::GenerateFile(const GeneratorOptions& options,
       // may be blocked by things like CSP.
       //   Function('') is almost the same as eval('')
       printer->Print(
-          "var global = (function() { return this || window || global || self "
-          "|| Function('return this')(); }).call(null);\n\n");
+          "var global = (function() {\n"
+          "  if (this) { return this; }\n"
+          "  if (typeof window !== 'undefined') { return window; }\n"
```

Comment:
we should add globalThis to the set of symbols to check for and check for that one first

Reply:
Done!